### PR TITLE
SUS-2983 | WikiFactory::getWikiByID - do not make queries when provided city_id will not return any row

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-2983

Number of queries to be saved: **1.9 mm daily on shared DB**

## Example

```php
> var_dump( WikiFactory::getWikiByID(0) )
bool(false)

> var_dump( WikiFactory::getWikiByID(5915) )
memcached: get(wikifactory:wikia:v1:5915)
memcached: MemCache: sock i:0; got wikifactory:wikia:v1:5915
object(stdClass)#288 (30) {
  ["city_id"]=>
  string(4) "5915"
  ["city_path"]=>
  string(5) "slot1"
  ["city_dbname"]=>
```

### CODE REVIEW

File: extensions/wikia/WikiaLogo/WikiaLogoHelper.class.php [hunk 1]
```diff
@@ -47,16 +47,14 @@ public function getCentralUrlForLang( $lang ) {
 	 * @return bool|GlobalTitle
 	 */
 	public function getCentralWikiUrlForLangIfExists( $lang ) {
-		try {
-			return GlobalTitle::newMainPage( $this->wikiCorporateModel->getCorporateWikiIdByLang( $lang ) );
-		} catch ( Exception $ex ) {
-			return false;
-		}
+		$city_id = $this->wikiCorporateModel->getCorporateWikiIdByLang( $lang );
+		return $city_id ? GlobalTitle::newMainPage( $city_id ) : false;
```
Comment (WikiaTech) [inline extensions/wikia/WikiaLogo/WikiaLogoHelper.class.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/WikiaLogo/WikiaLogoHelper.class.php [hunk 2]
```diff
@@ -47,16 +47,14 @@ public function getCentralUrlForLang( $lang ) {
 	 * @return bool|GlobalTitle
 	 */
 	public function getCentralWikiUrlForLangIfExists( $lang ) {
```
Comment (ghost) [inline extensions/wikia/WikiaLogo/WikiaLogoHelper.class.php:49]: Does this method still return anything?  I don't think we have corporate wikis. ?

**Review by ghost — APPROVED** [review]

File: extensions/wikia/WikiaLogo/WikiaLogoHelper.class.php (hunk 2 above)
Comment (macbre) [reply extensions/wikia/WikiaLogo/WikiaLogoHelper.class.php:49]: Unfortunately, yes - we still have them. Their domains are used to link to WordPress-powered sites.

**Pull request merged**